bundles/org.openhab.binding.orbitbhyve/src/main/java/org/openhab/binding/orbitbhyve/internal/handler/OrbitBhyveSprinklerHandler.java

```diff
@@ -0,0 +1,261 @@
+import org.eclipse.jdt.annotation.Nullable;
+import org.openhab.binding.orbitbhyve.internal.model.OrbitBhyveDevice;
+import org.openhab.binding.orbitbhyve.internal.model.OrbitBhyveDeviceStatus;
+import org.openhab.binding.orbitbhyve.internal.model.OrbitBhyveProgram;
+import org.openhab.binding.orbitbhyve.internal.model.OrbitBhyveZone;
+import org.openhab.core.library.types.DateTimeType;
+import org.openhab.core.library.types.DecimalType;
+import org.openhab.core.library.types.OnOffType;
+import org.openhab.core.library.types.StringType;
+import org.openhab.core.thing.Bridge;
+import org.openhab.core.thing.Channel;
+import org.openhab.core.thing.ChannelUID;
+import org.openhab.core.thing.Thing;
+import org.openhab.core.thing.ThingStatus;
+import org.openhab.core.thing.ThingStatusDetail;
+import org.openhab.core.thing.ThingStatusInfo;
+import org.openhab.core.thing.binding.BaseThingHandler;
+import org.openhab.core.thing.binding.builder.ChannelBuilder;
+import org.openhab.core.thing.binding.builder.ThingBuilder;
+import org.openhab.core.thing.type.ChannelTypeUID;
+import org.openhab.core.types.Command;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * The {@link OrbitBhyveSprinklerHandler} is responsible for handling commands, which are
+ * sent to one of the channels.
+ *
+ * @author Ondrej Pecta - Initial contribution
+ */
+@NonNullByDefault
+public class OrbitBhyveSprinklerHandler extends BaseThingHandler {
+
+    private final Logger logger = LoggerFactory.getLogger(OrbitBhyveSprinklerHandler.class);
+
+    public OrbitBhyveSprinklerHandler(Thing thing) {
+        super(thing);
+    }
+
+    private int wateringTime = 5;
+    private HashMap<String, OrbitBhyveProgram> programs = new HashMap<>();
+    private String deviceId = "";
+
+    @Override
+    public void handleCommand(ChannelUID channelUID, Command command) {
+        OrbitBhyveBridgeHandler handler = getBridgeHandler();
+        if (handler != null) {
+            if (CHANNEL_CONTROL.equals(channelUID.getId()) && command instanceof OnOffType) {
+                String mode = OnOffType.ON.equals(command) ? "auto" : "off";
+                handler.changeRunMode(deviceId, mode);
+                return;
+            }
+            if (CHANNEL_SMART_WATERING.equals(channelUID.getId()) && command instanceof OnOffType) {
+                boolean enable = OnOffType.ON.equals(command);
+                handler.setSmartWatering(deviceId, enable);
+                return;
+            }
+            if (!channelUID.getId().startsWith("enable_program") && OnOffType.OFF.equals(command)) {
+                handler.stopWatering(deviceId);
+                return;
+            }
+            if (CHANNEL_WATERING_TIME.equals(channelUID.getId()) && command instanceof DecimalType) {
+                wateringTime = ((DecimalType) command).intValue();
+                updateState(CHANNEL_WATERING_TIME, (DecimalType) command);
+                return;
```

Comment:
If you make this Number:Time, the command is of type `QuantityType`. As the user can send a command with any time unit, you need to convert the value to the the unit you expect with `toUnit()`. Take care that the return value can be null, if the unit cannot be converted.

Reply:
ok, fixed